pkg/registry/flowcontrol/ensurer/strategy.go

```diff
@@ -267,36 +269,37 @@ func EnsureConfiguration[ObjectType configurationObjectType](ctx context.Context
 
 		// we always re-create a missing configuration object
 		if _, err = ops.Create(ctx, ops.DeepCopy(bootstrap), metav1.CreateOptions{FieldManager: fieldManager}); err == nil {
-			klog.V(2).InfoS(fmt.Sprintf("Successfully created %T", bootstrap), "type", configurationType, "name", name)
+			logger.V(2).Info(fmt.Sprintf("Successfully created %T", bootstrap))
 			return nil
 		}
 
 		if !apierrors.IsAlreadyExists(err) {
 			return fmt.Errorf("cannot create %T type=%s name=%q error=%w", bootstrap, configurationType, name, err)
 		}
-		klog.V(5).InfoS(fmt.Sprintf("Something created the %T concurrently", bootstrap), "type", configurationType, "name", name)
+		logger.V(5).Info(fmt.Sprintf("Something created the %T concurrently", bootstrap))
 	}
 
-	klog.V(5).InfoS(fmt.Sprintf("The %T already exists, checking whether it is up to date", bootstrap), "type", configurationType, "name", name)
+	logger.V(5).Info(fmt.Sprintf("The %T already exists, checking whether it is up to date", bootstrap))
 	newObject, update, err := strategy.ReviseIfNeeded(ops, current, bootstrap)
 	if err != nil {
 		return fmt.Errorf("failed to determine whether auto-update is required for %T type=%s name=%q error=%w", bootstrap, configurationType, name, err)
 	}
 	if !update {
-		if klogV := klog.V(5); klogV.Enabled() {
-			klogV.InfoS("No update required", "wrapper", bootstrap.GetObjectKind().GroupVersionKind().Kind, "type", configurationType, "name", name,
-				"diff", cmp.Diff(current, bootstrap))
+		if loggerV := logger.V(5); loggerV.Enabled() {
+			loggerV.Info("No update required", "wrapper", bootstrap.GetObjectKind().GroupVersionKind().Kind)
 		}
 		return nil
 	}
 
-	if _, err = ops.Update(ctx, newObject, metav1.UpdateOptions{FieldManager: fieldManager}); err == nil {
-		klog.V(2).Infof("Updated the %T type=%s name=%q diff: %s", bootstrap, configurationType, name, cmp.Diff(current, bootstrap))
+	if updated, err := ops.Update(ctx, newObject, metav1.UpdateOptions{FieldManager: fieldManager}); err == nil {
+		if loggerV := logger.V(3); loggerV.Enabled() {
+			loggerV.Info(fmt.Sprintf("Updated the %T", bootstrap), "diff", cmp.Diff(current, updated))
```

Comment:
what does the unstructured `Info()` call do with the logger context values from earlier (`...WithValues("type", configurationType, "name", bootstrap.GetName())`? do we need to repeat those or do they get concatenated somehow?

Reply:
actually... is this still using structured logging in a way that makes the diff mentioned above unreadable? I can't really tell...

Reply:
Yes, this is a `logr.Logger`, whose `Info` method does structured logging. The presence of keys as well as values among the arguments, and absence of formatting directives for the arguments, is consistent with that.

So, yes, unfortunately, this is the structured logging that discards the linebreaks in the string returned from `Diff`, making it unreadable.

Reply:
> So, yes, unfortunately, this is the structured logging that discards the linebreaks in the string returned from Diff, making it unreadable.

No, I changed that already a while ago exactly for this reason: https://github.com/kubernetes/klog/pull/273

Reply:
Oh, good news! Thanks!